spec/enumerables_spec.rb

```diff
@@ -0,0 +1,151 @@
+            end
+        end
+    end
+
+    describe '#my_any?' do
+        context 'When a block is provided' do
+            it 'returns true when any of the number in the array is true' do
+                expect(arr.my_any? { |i| i.class == Integer }).to eql(true)
+            end
+
+            it 'returns true when any of the number in the array is true' do
+                expect(arr.my_any? { |i| i.class != Regexp }).to eql(true)
+            end
+        end
+
+        context 'When no block is given but arguments' do
+
+            it 'returns true when the class of one of the items is similar to the class for the argument' do
+                expect(arr.my_any?(arg == 23)).to eql(false)
+            end
+        end
+    end
+
+    describe '#.my_none?' do
+
+        context 'When array and block are given'do
+            it 'returns false when all the items in the array are true' do
+                expect( arr.my_none? { |i| i.class == Integer }).to eql(false)
+            end
+
+            it 'returns true when all the items in the array are not true' do
+                expect( arr.my_none? { |i| i.even? }).to eql(false)
+            end
+
+            it 'returns false when the argument is not regex' do
+                expect( arr.my_none? (arg.class == Regexp)).to eql( true )
+            end
+        end
+        context 'When no block is given' do
+            it 'returns false when the argument is an integer' do
+                expect( arr.my_none?( arg == 23 )).to eql( true )
+            end
+        end
+    end
+
+    describe '#my_map' do
+        context 'When a block is provided' do
+            it 'creates a new empty array to receive the items from the other array' do
+                expect(arr.my_map { |i| i }).to eql([1, 2, 3, 4, 3])
+            end
+
+            it 'returns true and false items in a new with ternary operators ' do
+                expect(arr.my_map { |i| i.even? }).to eql([false, true, false, true, false])
+            end
+        end
+
+        context 'When no block is given but arguments' do
+
+            it 'returns true when the class of one of the items is similar to the class for the argument' do
+                expect(arr.my_map).to be_instance_of(Enumerator)
+            end
+        end
+    end
+
+    describe '#my_count' do
+        context 'When a block is given' do
+            it 'counts the items in the array and returns total' do
+                expect(arr.my_count { |i| i < 4 }).to eql(4)
+            end
+
+        end
+
+        context 'When no block is given but arguments' do
+
+            it 'returns a count of items in the array basing on the input of the argument' do
+                expect(arr.my_count(arg)).to eql(0)
+            end
+
+            it 'returns a count of items in the array basing on the input of the argument' do
+                expect(arr.my_count{ |i|(arg > i)}).to eql(5)
+            end
```

Comment:
The context says when no block is given but the test you are checking contains a block. Isn't it?
Lets add another test case. When no block or argument is given it should count the number of elements in the array.
```ruby
it 'should return the number of elements in the array' do
  expect(arr.my_count).to eq(arr.count)
end